src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs

```diff
@@ -259,6 +272,25 @@ private void Initialize()
                 throw;
             }
         }
+
+        private enum FileSizeUnit
+        {
+            Byte, KB, MB, GB, TB, PB, EB
+        }
+
+        internal static string DisplayHumanReadableFileSize(long bytes)
+        {
+            if (bytes <= 0)
+            {
+                return $"0 {FileSizeUnit.Byte}";
+            }
+
+            var fileSizeUnit = (FileSizeUnit)(Math.Log10(bytes) / 3);
+            double value = bytes / (double)Math.Pow(1024, (long)fileSizeUnit);
+            var significantDigits = (int)fileSizeUnit;
+            var format = "0." + new string('0', significantDigits);
+            return value.ToString(format) + $" {fileSizeUnit}";
```

Comment:
I think this is overkill - we should do fewer allocations and, maybe, use integer arithmetic.

Reply:
As explained in my other comment [here](https://github.com/PowerShell/PowerShell/pull/14611#discussion_r558571942), I think the impact of this is negligible and borderline measurable.

Reply:
If we call this code millions of times, we get tens of megabytes of allocations. I still think we should be concerned about performance.

Reply:
I disagree because it is peanuts, something that is a fraction of a percent. I could throttle it to something where it only gets re-computed when the value changes by more than `1/100,000` but even that is overkill IMHO, Also, time in such scenarios is bound by download speed and the networking pressure on the CPU due to that is magnitudes higher. I will wait for another opinion on this matter.

Reply:
if the concern is reallocation, this can also be solved by having a fixed array of predefined format strings since the size units are fixed

Reply:
It is whole too expensive for the code we hardly optimized.
We could replace Log10 with switch, Pow with shift, make the format more simple.

Reply:
It would probably be simpler to just have a bunch of `if` statements comparing the size and returning the string instead of doing any math at all.

Reply:
- I see we download by chunks StreamHelper.ChunkSize = 10000. It is ~10Kb. (I wonder why don't we use exactly 10Kb?)
So from "Byte, KB, MB, GB, TB, PB, EB" we should remove Byte and KB.
- PB, EB - it is fantastic to have - we should remove too.
- For TB - it is good to use MB.

So we need only one static format with MB like "{0:0.00} MB".

Reply:
Ok, I adapted the code to use a switch statement with hard coded ranges and different significant digits based on the unit in my latest commit. This should resolve the performance concern.
https://github.com/PowerShell/PowerShell/pull/14611/commits/c4b6ba603810c65af83fb7b1ff3868c587f21978